### DISCUSSION TITLE
Can hocuspocus emit custom events?

### DISCUSSION THREAD

**Discussion by huanghantao** [ueberdosis/hocuspocus#815] [Q&A]
Can hocuspocus emit custom events? Like socketio. Currently, I have a requirement where user needs to set permissions for a document using an event called `SET_FILE_PERMISSION`, and then broadcast the document's permission list to other users in the room. Can I reuse the y-websocket connection for this or do I have to create a separate websocket connection to handle events other than document collaboration? Is y-websocket only used for collaboration purposes?

**Comment by grootgordon** [comment]
Yes! We can use `sendStateless` method that defined in `class HocuspocusProvider`  on client side.
And handle it  by `onStateless` in server side. 
refer: https://tiptap.dev/docs/hocuspocus/server/hooks#on-stateless

**Comment by codingdeng** [comment]
通过打压我获取低彩礼，然后再骗人说十万爸妈要借钱，对我一顿怒火，再放大小事，抓把柄拿对父母不尊重的名义说事，然后利用我产生的愧疚把责任全部推卸给我，再全身而退，呸，你怎么这么不要脸，好歹处了这么久，我能被你pua，自责这么久，还不是因为有感情被你拿捏，你居然下得去手，真的贱，订婚三个月你那连邻居都不知道，我那人尽皆知 @huanghantao
